widget/entry.go

```diff
@@ -960,13 +977,22 @@ func (r *entryRenderer) Objects() []fyne.CanvasObject {
 
 // Refresh satisfies the fyne.WidgetRenderer interface.
 func (r *entryRenderer) Refresh() {
-	if r.entry.Text != string(r.entry.textProvider().buffer) {
-		r.entry.textProvider().SetText(r.entry.Text)
+	var provider, placeholder *textProvider
+	content := r.entry.GetField(func() interface{} {
+		provider = r.entry.textProvider()
```

Comment:
This could create the textProvider without holding a write lock

Reply:
are you sure? - textProvider() can write the field if there is no provider...

Reply:
But it would write the field while holding the read lock, but not the write lock

Reply:
It is my understanding that if you have RWMutex.Lock() then nobody can .RLock() I don't think that Lock() is write-only